ecdsa_private_key.go

```diff
@@ -0,0 +1,205 @@
+package hedera
+
+import (
+	"bytes"
+	"crypto/ecdsa"
+	"crypto/elliptic"
+	"crypto/rand"
+	"encoding/hex"
+	"fmt"
+	"math/big"
+	"strings"
+
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/hashgraph/hedera-protobufs-go/services"
+	"github.com/pkg/errors"
+)
+
+// _ECDSAPrivateKey is an Key_ECDSASecp256K1 private key.
+type _ECDSAPrivateKey struct {
+	*ecdsa.PrivateKey
+}
+
+func _GenerateECDSAPrivateKey() (*_ECDSAPrivateKey, error) {
+	key, err := ecdsa.GenerateKey(crypto.S256(), rand.Reader)
+	if err != nil {
+		return &_ECDSAPrivateKey{}, err
+	}
+
+	return &_ECDSAPrivateKey{
+		key,
+	}, nil
+}
+
+func _ECDSAPrivateKeyFromBytes(byt []byte) (*_ECDSAPrivateKey, error) {
+	length := len(byt)
+	switch length {
+	case 32:
+		return _RawECDSAPrivateKeyFromBytes(byt)
+	case 65:
+		return _DerECDSAPrivateKeyFromBytes(byt)
+	default:
+		return &_ECDSAPrivateKey{}, _NewErrBadKeyf("invalid private key length: %v bytes", len(byt))
+	}
+}
+
+func _RawECDSAPrivateKeyFromBytes(byt []byte) (*_ECDSAPrivateKey, error) {
+	length := len(byt)
+	if length != 32 {
+		return &_ECDSAPrivateKey{}, _NewErrBadKeyf("invalid private key length: %v bytes", len(byt))
+	}
+
+	b := bytes.Trim(byt, "\x00")
```

Comment:
I don't think you should be trimming `\x00`'s from the start. Just `0x`

Reply:
I am trimming empty bytes here, because as I saw in how the private key is made it can have empty bytes in order to make it len 32.